### ISSUE TITLE
Android Arch Components handle configuration changes / lifecycle 

### ISSUE THREAD

**Issue by ashrafi** [android10/Android-CleanArchitecture-Kotlin#63]
you said 

> At implementation level, in our example, MVVM is accomplished by the usage of Architecture Components, which its main advantage is to handle configuration changes when the screen rotates, something that has given us many headaches as android developers (I guess you know what I’m talking about).
> 
> **Disclaimer: that does not mean we have to no longer care about lifecycles, but it is way easier.**

ViewModel was created to handle the configuration / lifecycle for you.

[viewModel](https://developer.android.com/topic/libraries/architecture/viewmodel)
[viewModel Lifecycle](https://developer.android.com/reference/android/arch/lifecycle/ViewModel)

**Comment by Zhuinden** [comment]
Don't forget about onSaveInstanceState though

**Comment by ashrafi** [comment]
Good point but build in a way that I no longer need to use onSaveInstanceState :-)

> Additionally, when you open an activity from an intent, the bundle of extras is delivered to you on both configuration changes and when the system restores an activity. If the search query were passed in as an intent extra, you could use the extras bundle instead of the onSaveInstanceState() bundle.

https://medium.com/androiddevelopers/viewmodels-persistence-onsaveinstancestate-restoring-ui-state-and-loaders-fc7cc4a6c090

**Comment by Zhuinden** [comment]
You can't throw *every* possible transient state and user input into an Intent.putExtra, because the user will start doing things *after* you open the Activity. 🤔

**Comment by St4B** [comment]
Also, the bundle has a size limit of about 1MB, which is not a problem if you are not going to put lists of objects. The main idea was to save ids and call again again the needed actions in order to retrieve your state.

Check this version of mvp that i had created. The presenter keeps the annotated data even if the app is killed by the OS. In addition to this, you do not need to stop a task an restart it during configuration changes. Maybe you can use a combination of ViewModel with my mvp in order to achieve what you expect.

You can check it here: https://github.com/St4B/mvp
